packages/sns/lib/utils/stsUtils.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  GetCallerIdentityCommand,
+  type GetCallerIdentityCommandOutput,
+  type STSClient,
+} from '@aws-sdk/client-sts'
+
+let callerIdentityPromise: Promise<void> | undefined
+let callerIdentityCached: GetCallerIdentityCommandOutput | undefined
+
+/**
+ * Manually builds the ARN of a topic based on the current AWS account and the topic name.
+ * It follows the following pattern: arn:aws:sns:<region>:<account-id>:<topic-name>
+ * Doc -> https://docs.aws.amazon.com/IAM/latest/UserGuide/reference-arns.html
+ */
+export const buildTopicArn = async (client: STSClient, topicName: string) => {
+  const identityResponse = await getAndCacheCallerIdentity(client)
+  const region =
+    typeof client.config.region === 'string' ? client.config.region : await client.config.region()
+
+  return `arn:aws:sns:${region}:${identityResponse.Account}:${topicName}`
+}
+
+export const clearCachedCallerIdentity = () => {
```

Comment:
let's expose it from package exports as well, because users may need it too

Reply:
Good point! I forgot to expose it on index 😓